docs/content/preview/api/ysql/the-sql-language/statements/ddl_alter_table.md

```diff
@@ -198,6 +198,50 @@ It quietly succeeds. Now `\d children` shows that the foreign key constraint `ch
 
 Add the specified constraint to the table.
 
+#### [*alter_column_type*]
+
+Change the type of an existing column. If data on disk is required to change, a full table rewrite is needed.
+
+The new column type will be applied automatically to indexes and simple table constraints by reanalyzing the initially provided expression.
+
+If the optional `COLLATE` clause is not specified, the default collation for the new column type will be used.
+
+If the optional `USING` clause is not provided, the default conversion for the new column value will be the same as an assignment cast from the old type to the new type.
+
+A `USING` clause must be included when there is no implicit or assignment cast available from the old type to the new type.
+
+##### Alter type without table-rewrite
+
+If the change doesn't require data on disk to change, these semantics apply:
+
+- Concurrent DMLs to the table can go through.
+
+Example:
+
+```sql
+CREATE TABLE test (id BIGSERIAL PRIMARY KEY, a VARCHAR(50));
+ALTER TABLE test ALTER COLUMN a TYPE VARCHAR(51);
+```
+
+##### Alter type with table rewrite
+
+If the change requires data on disk to change, a full table rewrite will be done and the following semantics apply:
+
+- This action is offline, so concurrent DMLs should not be performed while an alter column type operation is happening.
```

Comment:
@ddorian / @ddhodge can we also document this "unsafe" behavior for ALTER ADD/DROP pkey?